lib/core/src/Cardano/Wallet/TokenMetadata.hs

```diff
@@ -401,14 +402,29 @@ nullTokenMetadataClient :: Applicative m => TokenMetadataClient m
 nullTokenMetadataClient = TokenMetadataClient $ \_ ->
     pure . Right $ BatchResponse []
 
+-- | Mock metadata client using a hard-coded server response from a file for
+-- data.
+mockFileTokenMetadataClient :: URI -> TokenMetadataClient IO
+mockFileTokenMetadataClient uri = TokenMetadataClient $ \(BatchRequest subjects _)-> do
+    let filePath = T.unpack
+            . fromMaybe (error "mockFileTokenMetadataClient: bad file path")
+            . T.stripPrefix "file:/" . T.pack $ show uri
+    BatchResponse mds <- either (error . show) id
+        <$> eitherDecodeFileStrict filePath
```

Comment:
This would decode the file on each request. I guess that's okay in this context and could actually be useful to dynamically change the content of the mock server :thinking: